Marlin/Marlin_main.cpp

```diff
@@ -1806,13 +1809,9 @@ static void clean_up_after_endstop_or_probe_move() {
       }
     #endif
 
-    float z_dest = LOGICAL_Z_POSITION(z_raise);
+    float z_dest = z_raise;
```

Comment:
I haven't had an opportunity to test this on my cartesian machine, but I believe this will work as expected. However, it would be good if someone could verify.